UnitySDK/Assets/ML-Agents/Scripts/Sensor/CameraSensor.cs

```diff
@@ -0,0 +1,46 @@
+using System;
+using MLAgents.InferenceBrain;
+using UnityEngine;
+
+namespace MLAgents.Sensor
+{
+    public class CameraSensor : SensorBase
```

Comment:
Why is this implementing a SensorBase and not a ISensor?

Reply:
If it is to get the monobehavior, it could inherit `ISensor, MonoBehavior`

Reply:
Yeah, it could do that. But my plan for Agent setup was to call `GetComponents<SensorBase>()` (and allow the user to provide their own list of ISensors if they have custom implementations)

Reply:
Ah, and that requires SensorBase to inherit MonoBehaviour, I see.

Reply:
I'm open to other suggestions, but that definitely seemed like the easiest way to allow setting up the sensors in the editor.

Reply:
Still I would have : 
```csharp
interface ISensor {}
abstract SensorComponent : ISensor, MonoBehavior  {}//(is a component and a ISensor and can use GetComponents<SensorComponent>())
abstract SensorBase : ISensor  {}// Not a component but is provides a simplified way to collect observations
```

Reply:
But then what about RayPerception, which we want to be a SensorComponent (it's already a MonoBehaviour) but want to produce floats for observations? It would be OK if we put the float observations in ISensor.

Reply:
We could have SensorComponent as a holder for an ISesnor if we want to avoid having ISensors all implanted as MonoBehaviours.

Reply:
Updated, SensorComponent now produces an ISensor.